libbeat/logp/file_rotator.go

```diff
@@ -54,7 +55,11 @@ func (rotator *FileRotator) CheckIfConfigSane() error {
 	}
 
 	if *rotator.KeepFiles < 2 || *rotator.KeepFiles >= RotatorMaxFiles {
-		return fmt.Errorf("The number of files to keep should be between 2 and %d", RotatorMaxFiles-1)
+		return fmt.Errorf("the number of files to keep should be between 2 and %d", RotatorMaxFiles-1)
+	}
+
+	if rotator.Permissions != nil && (*rotator.Permissions > 07777) {
```

Comment:
Can you replace the `07777` with the constant `os.ModePerm` (which is `0777`). That should be the max value because users should be setting mode type.